tests/forms_tests/tests/test_forms.py

```diff
@@ -1369,6 +1370,24 @@ class UserRegistration(Form):
         self.assertEqual(bound['password'].value(), 'foo')
         self.assertEqual(unbound['password'].value(), None)
 
+    def test_boundfield_initial_called_once(self):
+        """
+        Multiple calls to BoundField().value() in an unbound form should return
+        the same result each time (#24391).
+        """
+        class MyForm(Form):
+            name = CharField(max_length=10, initial=uuid.uuid4)
+
+        form = MyForm()
+        name = form['name']
+        val1 = name.value()
+        val2 = name.value()
+        self.assertEqual(val1, val2)
+        # BoundField itself is also cached
+        name = form['name']
+        val3 = name.value()
+        self.assertEqual(val2, val3)
```

Comment:
Shouldn't that be `self.assertIs(name, form['name'])` to check for the same instance. No need for `val3` then.